server/src/test/java/org/opensearch/index/shard/SegmentReplicationIndexShardTests.java

```diff
@@ -965,6 +1035,30 @@ public void testReuseReplicationCheckpointWhenLatestInfosIsUnChanged() throws Ex
         }
     }
 
+    public void testComputeReplicationSegmentCheckpoint() throws Exception {
```

Comment:
Would it be possible in these test to create a hook at the warm method by overriding the IndexWarmer with a custom IndexWarmer?